rear_rider_device/leds_proc.py

```diff
@@ -134,37 +110,50 @@ async def on_is_strobe_on(self):
         )
     
     async def on_strobe_off(self):
-        # self.writeline(
-        #     'strobe_off_ack'
-        # )
-        await self._join_strobe_thread()
-        # self.writeline(
-        #     'strobe_off_ok'
-        # )
+        self._leds_effects_loop_ctx.set_effects([])
     
-    async def _join_strobe_thread(self):
-        # Set _strobe_on to False to tell the thread to stop
-        self._strobe_on = False
-
-        if self._strobe_task_thread is not None:
-            self.writeline('strobe_task_waiting')
-            self._strobe_task_thread.join()
-            self._strobe_task_thread = None
+    def _join_leds_thread(self):
+        if self._leds_effects_loop_thread is not None:
+            self._leds_effects_loop_ctx.stop()
+            self._leds_effects_loop_thread.join()
+            self._leds_effects_loop_thread = None
     
     async def on_discoverable_on(self):
-        pass
+        self.led_strip.set_brightness(0.25)
+        self._set_strobe_effect(1, DISCOVERABLE_EFFECT_COLOR)
 
     async def on_discoverable_off(self):
-        pass
+        self._leds_effects_loop_ctx.set_effects([])
+        self.led_strip.turn_off()
     
-    async def on_add_effect(self):
+    async def on_set_effect(self):
         params = (await self.readline()).split(' ')
         pattern = params[0]
         brightness = params[1]
+        """brightness
+        1 - low
+        2 - medium
+        3 - high
+        """
         color = (int(params[2]), int(params[3]), int(params[4]))
         if pattern == '1':
-            self.led_strip.set_brightness(float(brightness))
-            await self._start_strobe_thread(5, 0, color)
+            self.led_strip.set_brightness(1.0 / (4 - int(brightness)))
+            self._set_strobe_effect(5, color)
+        else:
+            return
```

Comment:
food for thought:

swapping this `if/else` around imo would make for better readability and flow:

```
if pattern != '1':
    return
// continue with logic
```

it's also more performant since you're cutting down on jumps and checks. 

granted, when more patterns are introduced this all goes out the window, and you'll need a way to differentiate logic based on the value. but i don't think we'll implement more by the end of the semester lol

Reply:
I agree with this. I usually try to do what you brought up but there are times where I totally forget to flip the logic.